crates/libcontainer/src/tty.rs

```diff
@@ -75,12 +75,21 @@ pub fn setup_console_socket(
     container_dir: &Path,
     console_socket_path: &Path,
     socket_name: &str,
-) -> Result<RawFd> {
+) -> Result<OwnedFd> {
+    struct CurrentDirGuard {
+        path: PathBuf,
+    }
+    impl Drop for CurrentDirGuard {
+        fn drop(&mut self) {
+            let _ = env::set_current_dir(&self.path);
+        }
+    }
```

Comment:
just comment: I haven't seen this pattern before, seems a good way to manage this kind of thing locally!

Reply:
I think there is a crate called `scopeguard`, which contains a `defer!` macro, can handle this more elegantly. I just don't want to introduce the dependency.